addons/stock/models/stock_picking.py

```diff
@@ -975,10 +975,7 @@ def button_validate(self):
         return True
 
     def action_set_quantities_to_reservation(self):
-        for move in self.move_lines.filtered(lambda move: move.state in ('partially_available', 'assigned')):
-            if move.product_id.tracking == 'none':
-                for move_line in move.move_line_ids:
-                    move_line.qty_done = move_line.product_uom_qty
+        self.move_lines._set_reservation_as_done()
```

Comment:
Detail but I would also rename the action name or this function name to have consistency